### ISSUE TITLE
How can reuse instance of created component

### ISSUE THREAD

**Issue by adryweber** [react-figma/react-figma#199]
Hi, can i programmably reuse component via call his instance?

https://react-figma.now.sh/src/components/component/Component.html

<img width="1369" alt="Снимок экрана 2019-11-28 в 20 27 09" src="https://user-images.githubusercontent.com/15189959/69824648-813f9200-121d-11ea-9473-cb3bd46cda7c.png">

**Comment by ilyalesik** [comment]
I think we should implement API for creating instances, but right now it's not supported. I don't know how to design React-way API for it. [react-sketchapp](http://airbnb.io/react-sketchapp/docs/API.html#symbols) Symbols API not looks so good. Maybe you have ideas on how API for instance creation can look?

**Comment by ilyalesik** [comment]
It's possible to implement API similar to React Context:

```javascript
export const HeadingH1 = createComponent();

export const Styleguide = () => {
    return (
        <Frame name="Styleguide">
            <HeadingH1.Component>
                <Text>Heading H1</Text>
            </HeadingH1.Component>
         </Frame>
     )
}

...

export const Example = () => {
  return <HeadingH1.Instance />
}
```

But some of API for overrides content inside the instance still needed.

**Comment by adryweber** [comment]
I used the sketch-app and created a lot of Symbols there (in fact components). Also override them a lot. In Sketch a had over 300 Symbols. It was quite convenient and understandable.

http://airbnb.io/react-sketchapp/docs/API.html#symbols

**Comment by adryweber** [comment]
Unfortunately, now i have no idea for better API for this. I'm not full programmer :)

**Comment by ilyalesik** [comment]
Great news! Instance API is available (since `0.0.34`).

### Examples:

#### Creating component instance

```javascript
const Rect = createComponent();

...

<>
    <Rect.Component name="rect-component">
        <Rectangle
            style={{
                width: 200,
                height: 100,
                backgroundColor: "blue"
            }}
        />
    </Rect.Component>
    <Rect.Instance style={{ marginTop: 20, width: '100%' }} />
</>
```

### Overriding elements inside instance

```javascript
const Rect = createComponent();

...

<>
    <Rect.Component name="rect-component">
        <Rectangle
            name="rect"
            style={{
                width: 200,
                height: 100,
                backgroundColor: "blue"
            }}
        />
    </Rect.Component>
    <Rect.Instance style={{ marginTop: 20, width: '100%' }} overrides={{
        rect: {
            style: {
                backgroundColor: "red"
            }
        } 
    }} />
</>
```

Text content overriding:

```javascript
const Txt = createComponent();
...
<Txt.Component name="text-component" style={{ marginTop: 20 }}>
    <Text name="text" style={{ fontSize: 24, width: 200 }}>
        Some text
    </Text>
</Txt.Component>
<Txt.Instance
    style={{ marginTop: 20 }}
    overrides={{
        text: {
            characters: 'Overrided text'
        }
    }}
/>
```
